File: problems/intermediate/ema_supercomputer/__tests__/test_ema_supercomputer..js
const { twoPluses } = require("../ema_supercomputer");

const grid = [
    "GGGGGG",
    "GBBBGB",
    "GGGGGG",
    "GGBBGB",
    "GGGGGG"
];

const result = twoPluses(grid);

if (result !== 5) {
    throw new Error(`❌ Test failed. Expected 5, got ${result}`);
}

console.log("✅ JavaScript test passed!");

File: problems/intermediate/ema_supercomputer/__tests__/test_ema_supercomputer..cpp
#include <iostream>
#include <vector>
#include <string>
#include "../ema_supercomputer.cpp"
using namespace std;

void test() {
    vector<string> grid = {
        "GGGGGG",
        "GBBBGB",
        "GGGGGG",
        "GGBBGB",
        "GGGGGG"
    };

    int result = twoPluses(grid);

    if (result != 5) {
        cout << "❌ Test failed. Expected 5, got " << result << "\n";
    } else {
        cout << "✅ C++ test passed!\n";
    }
}

int main() {
    test();
    return 0;
}
